libs/backend/src/cast_vote_records/export.ts

```diff
@@ -423,6 +425,76 @@ async function exportSignatureFileToUsbDrive(
   return ok();
 }
 
+/**
+ * Updates the creation timestamp of a directory and its children files (assuming no
+ * sub-directories) using the one method guaranteed to work:
+ * ```
+ * cp -r <directory-path> <directory-path>-temp
+ * rm <directory-path>
+ * mv <directory-path>-temp <directory-path>
+ * ```
+ */
+async function updateCreationTimestampOfDirectoryAndChildrenFiles(
+  directoryPath: string
+): Promise<void> {
+  const tempDirectoryPath = `${directoryPath}-temp`;
+  await fs.mkdir(tempDirectoryPath);
+  const fileNames = (await fs.readdir(directoryPath, { withFileTypes: true }))
+    .filter((entry) => entry.isFile())
+    .map((entry) => entry.name);
+  for (const fileName of fileNames) {
+    await fs.copyFile(
+      path.join(directoryPath, fileName),
+      path.join(tempDirectoryPath, fileName)
+    );
+  }
+  await fs.rm(directoryPath, { recursive: true });
+  await fs.rename(tempDirectoryPath, directoryPath);
+}
+
+/**
+ * File creation timestamps could reveal the order in which ballots were cast. To maintain
+ * voter privacy, every time a ballot is cast, we randomly select 1 or 2 existing cast vote records
+ * and update their creation timestamps to the present. This approach is equivalent to taking
+ * random ballots from a stack and placing them on top every time a new ballot is added to the
+ * stack, a kind of shuffling as we go.
+ *
+ * This shuffling as we go is irrelevant for full exports, where we already iterate over cast vote
+ * records in an order independent of creation timestamp.
+ */
+async function maintainVoterPrivacy(
+  exportContext: ExportContext
+): Promise<void> {
+  const { usbMountPoint } = exportContext;
+
+  const exportDirectoryPath = path.join(
+    usbMountPoint,
+    getExportDirectoryPathRelativeToUsbMountPoint(exportContext)
+  );
+  const castVoteRecordIds = (
+    await fs.readdir(exportDirectoryPath, { withFileTypes: true })
+  )
+    .filter((entry) => entry.isDirectory())
+    .map((entry) => entry.name);
+
+  if (castVoteRecordIds.length < 2) {
+    return;
+  }
+
+  const oneOrTwo = Math.random() < 0.5 ? 1 : 2;
```

Comment:
Is there a security reason for randomly picking a number here?

My first thought is that it would have been based on the size of the pile - the more CVRs there are, the more should be reshuffled.

Reply:
Related: [`Math.random()` is not cryptographically secure](https://kemilbeltre.medium.com/why-do-not-use-math-random-a6f8b0ad38dd). We should probably add a lint rule against its use. You could do this instead:
```suggestion
  const oneOrTwo = getRandomValues(new Uint8Array(1))[0] % 2 === 1 ? 1 : 2;
```

Reply:
Good q @adghayes! You can basically make more inferences if you're always moving the same number of CVRs. For example, if you're always moving one CVR (and assuming we already have lots of CVRs), there's a high likelihood that every other CVR among the latest CVRs is one of the last ones cast. Even if you're always moving two CVRs, the same situation can arise, but with every third ballot instead of every other. Adding this additional source of randomness makes the whole system less orderly.

Somewhat relatedly, since opening this PR, I've been thinking about protecting the privacy of the last voter to cast their ballot before polls close. And I'm thinking of adding one more source of randomness to accomplish that, since there currently are situations where you could infer which CVR belongs to that voter.

Reply:
Thank you for the reminder @eventualbuddha! I recall you updating PIN generation code some time a while back for this same reason. I think that the fact that `Math.random` is not cryptographically secure will really stick this time, but I also agree that a lint rule is a good idea

Reply:
https://github.com/votingworks/vxsuite/pull/3972/commits/5785f26535d5c1d5158144da7dfb7d74b6134dcc